addons/web/static/src/js/views/list/list_editable_renderer.js

```diff
@@ -264,10 +265,59 @@ ListRenderer.include({
      */
     getEditableRecordID: function () {
         if (this.currentRow !== null) {
-            return this.state.data[this.currentRow].id;
+            return this.getRecordID(this.currentRow);
         }
         return null;
     },
+    /**
+     * for editable group by mode record could be deep down in state tree
+     * traverse state and get record
+     *
+     * @param {string} [recordID]
+     * @returns {object}
+     */
+    getRecord: function (recordID) {
+        var record;
+        utils.traverse_records(this.state, function (r) {
+            if (r.id === recordID) {
+                record = r;
+            }
+        });
+        return record;
+    },
+    /**
+     * retrieves recordID using rowIndex as DOM element table row stores recordID data
+     *
+     * @param {integer} rowIndex
+     * @returns {string} recordID
+     */
+    getRecordID: function (rowIndex) {
+        var $tr = this.$('table.o_list_view > tbody tr').eq(rowIndex);
+        return $tr.data('id');
+    },
+    /**
+     * get datapoint in form of flat array from state tree
+     *
+     * @returns {array}
+     */
+    getRecordIDList: function () {
```

Comment:
maybe not necessary (used by the navigation (ENTER) to move from a group to another)
if necesseary, we may move this to the datapoint directly (?)